core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -674,6 +677,18 @@ private[spark] class Executor(
     }
   }
 
+  private def setMDCForTask(taskDescription: TaskDescription): Unit = {
+    val properties = taskDescription.properties
+
+    org.slf4j.MDC.put("appId", properties.getProperty("spark.app.id"))
+    org.slf4j.MDC.put("appName", properties.getProperty("spark.app.name"))
+
+    properties.asScala.filter(_._1.startsWith("mdc.")).foreach { item =>
+      val key = item._1.substring(4)
+      org.slf4j.MDC.put(key, item._2)
```

Comment:
Can we use `org.apache.log4j` instead?

Reply:
If the only interface for user yet is task runnable, why do we have to touch the whole `ThreadUtils`?

Reply:
1. I think where possible it is better to use abstract layer - at least in new code.
2. about ThreadUtils without all that change the MDC will not propagate to all places only to the scope of that thread and from my tests, it is not enough, we are using that code internally for more than a year.

Reply:
IIUC, MDC values can only be propagated at executor side and only the threads which created after task threads could see the values. So I am really confused that how can you guarantee values can be propagated to all places?

Reply:
exactly, we want all the threads that handle work that connected to tasks will get MDC so their logs will be traceable.